src/main/java/org/opentripplanner/transit/model/network/SubMode.java

```diff
@@ -0,0 +1,113 @@
+package org.opentripplanner.transit.model.network;
+
+import java.io.Serializable;
+import java.util.BitSet;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class wrap a string and create an index for it, so we can use BitSet for matching a
+ * trip netexSubMode. SubModes are used in trip filtering for every request.
+ * <p>
+ * Naming, this class is named SubMode, not NetexSubMode because we want to migrate gtfsRouteType
+ * into the same concept.
+ * <p>
+ * This class is thread-safe, and the performance overhead should affect the graph built time,
+ * but not routing.
+ */
+public class SubMode implements Serializable {
+
+  private static final int NONE_EXISTING_SUBMODE_INDEX = 1_000_000;
+
+  private static final Logger LOG = LoggerFactory.getLogger(SubMode.class);
+  private static final AtomicInteger COUNTER = new AtomicInteger(0);
+  private static final Map<String, SubMode> ALL = new ConcurrentHashMap<>();
+
+  public static final SubMode UNKNOWN = getOrBuildAndCashForever("unknown");
```

Comment:
There is a typo: "cash" should be "cache".

Reply:
This is a common type of mine, I search and replaced the entire code and fixed it several places, not just here.